Dockerfile

```diff
@@ -0,0 +1,88 @@
+
+# This is a Dockerfile that extends an installation of itk and modules
+#in the dsarchive/histomicstk:v0.1.3 
+
+
+FROM dsarchive/histomicstk:v0.1.3
+
+MAINTAINER Bilal Salam <bilal.salam@kitware.com>
+
+
+RUN apt-get update && \
+    apt-get install -y git wget python-qt4\
+    openslide-tools python-openslide \
+    build-essential \
+    swig \
+    make \
+    zlib1g-dev \
+    curl \
+    libcurl4-openssl-dev \
+     libexpat1-dev \
+    unzip && \
+    apt-get autoremove && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+#install requirements for itk
+
+WORKDIR /
+
+#need cmake
+RUN git clone https://github.com/Kitware/CMake.git
+WORKDIR CMake 
+#cmake release v3.6.0
+RUN git checkout e31084e65745f9dd422c6aff0a2ed4ada6918805 && \
+./bootstrap --system-curl && make && make install
+#by default cmake does not use system ssl
+WORKDIR /
+
+#ITK dependencies
+
+RUN mkdir ninja
+WORKDIR ninja
+
+RUN wget https://github.com/ninja-build/ninja/releases/download/v1.7.1/ninja-linux.zip && \
+unzip ninja-linux.zip && \
+ln -s $(pwd)/ninja /usr/bin/ninja && \
+ninja --version
+
+WORKDIR /
+#need to get the latest tag of master branch in ITK
+# v4.10.0 = 95291c32dc0162d688b242deea2b059dac58754a
+RUN git clone https://github.com/InsightSoftwareConsortium/ITK.git
+WORKDIR ITK
+RUN git checkout $(git describe --abbrev=0 --tags)
+WORKDIR /
+RUN mkdir ITKbuild
+WORKDIR ITKbuild
+
+RUN cmake \
```

Comment:
All the above commands for cloning and checking out ITK can be combined with this RUN command as follows:

```Dockerfile
RUN cd / && git clone https://github.com/InsightSoftwareConsortium/ITK.git && cd ITK && \
    git checkout $(git describe --abbrev=0 --tags) && mkdir -p build && \
    cmake \
```